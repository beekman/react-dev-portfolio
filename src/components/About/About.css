@value colors: '../colors.css';
@value lightblue, yellow, offwhite, white, darkgray from colors;

.About {
  position: relative;
  height: auto;
  min-height: 100vh;
  color: white;
  background-color: darkgray;
  display: flex;
  top: 3rem;

  header, main {
    max-width: 1200px;
  }

  header, section, summary {
    max-width: 1200px;
  }

  p {
    color: offwhite;
  }
}

@media screen and (max-width: 720px){
  .About {
    flex-direction: column;
    top: 120px;
    padding: 10px;
    position: relative;


    header {


      h1 {
        margin: auto;
        text-align: center;
        font-size: 1.5rem;
        position: relative;
        color: #45E8FC;
        position: relative;
        top: 50px;
        color: yellow;
      }
    }

    main{
      display: flex;
      flex-direction: column-reverse;
      width: 100%;


      section{
        justify-content: center;
        align-items: center;
        padding: 1rem;
      }

      figure {
        margin: 4.5rem auto 0;

        img {
          border-radius: 50%;
          max-width: 80px;
        }
      }
    }
  }
}
@media screen and (min-width: 721px){
  .About {
    justify-content: space-evenly;
    align-items: center;
    flex-direction: column;
    min-height: 100vh;
    top: 3.2rem;
    padding: 1rem 2rem;
    header {

      h1 {
        margin: auto;
        text-align: center;
        font-size: 3rem;
        padding-top: 2rem;
        color: yellow;
      }
    }

    main{
      display: grid;
      grid-template-columns: 3fr 1fr;

      section{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        padding: 1rem;
      }

      figure {
        padding: 1rem;

        img {
          border-radius: 50%;
          max-width: 150px;
        }
      }
    }
  }
}
