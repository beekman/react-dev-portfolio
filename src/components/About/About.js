import React from 'react';
import PropTypes from 'prop-types';
import styles from './About.css';
import Typist from 'react-typist';
import { Element } from 'react-scroll';
import photo from '../../assets/ben-beekman-profile-800.jpg';

const About = ({ about }) => {
  return (

    <Element name="About" className={styles.About}>
      <header>
        <h1>
          <Typist cursor={{ hideWhenDone: true }}>
            Hi, I&apos;m Ben. I&apos;m a Creative Software Developer
            <Typist.Backspace count={18} delay={750} /><Typist.Delay ms={800} />Front End Developer<Typist.Backspace count={19} delay={750} /><Typist.Delay ms={850} />Back End Engineer<Typist.Backspace count={17} delay={500} /><Typist.Delay ms={1150} />Full Stack Developer<Typist.Backspace count={35} delay={500} />It&apos;s complicated.</Typist>
        </h1>
      </header>

      <main>
        <section>
          <p>{about.summary}</p>
          <p>{about.pitch}</p>
        </section>
        <figure className={styles.profile}>
          <img src={photo} />
        </figure>
      </main>
    </Element>
  );
};

About.propTypes = {
  about: PropTypes.shape({
    title: PropTypes.string.isRequired,
    summary: PropTypes.string.isRequired,
    pitch: PropTypes.string.isRequired
  })
};

export default About;
