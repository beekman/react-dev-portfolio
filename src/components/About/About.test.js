import React from 'react';
import { shallow } from 'enzyme';
import About from './About';
import data from '../../portfolio-data.js';

describe('About component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<About about={data.about}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
