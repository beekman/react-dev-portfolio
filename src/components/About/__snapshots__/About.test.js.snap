// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<Element
  className="About"
  name="About"
>
  <header>
    <h1>
      <Typist
        avgTypingDelay={70}
        className=""
        cursor={
          Object {
            "hideWhenDone": true,
          }
        }
        delayGenerator={[Function]}
        onCharacterTyped={[Function]}
        onLineTyped={[Function]}
        onTypingDone={[Function]}
        startDelay={0}
        stdTypingDelay={25}
      >
        Hi, I'm Ben. I'm a Full Stack Web Engineer
        <Backspace
          count={12}
          delay={750}
        />
        <Delay
          ms={800}
        />
        Software Engineer
        <Backspace
          count={28}
          delay={750}
        />
        <Delay
          ms={850}
        />
        Front End Engineer
        <Backspace
          count={18}
          delay={500}
        />
        <Delay
          ms={850}
        />
        Back End Engineer
        <Backspace
          count={17}
          delay={500}
        />
        <Delay
          ms={1150}
        />
        Full Stack Engineer.
      </Typist>
    </h1>
  </header>
  <main>
    <section>
      <p>
        I'm a full-stack web and software engineer with an entrepreneurial background in front-end web development, SEO, and education. I'm enthusiastic about creating code that solves human-centered problems and I love learning and teaching new techniques and technologies.
      </p>
      <p>
        As a full-stack web engineer role at a startup, I spent the past fifteen months working on a wide variety of web-based software and plugins to aid with internal multi-departmental processes. Before that, I spent more than a decade developing and designing websites.
      </p>
    </section>
    <figure
      className="profile"
    >
      <img
        src="/path/image"
      />
    </figure>
  </main>
</Element>
`;
